resources/migrations/000_migrations.yaml

```diff
@@ -9748,6 +9748,23 @@ databaseChangeLog:
         - onFail: MARK_RAN
         - dbms:
             type: mysql,mariadb
+  - changeSet:
+      id: v42.00-065
+      author: dpsutton
+      comment: >-
+        Added 0.42.0 - Another modal dismissed state on user. Retaining the same suffix and boolean style to ease an
+        eventual migration.
+      changes:
+        - addColumn:
+            tableName: core_user
+            columns:
+              - column:
+                  name: is_datasetnewb
```

Comment:
Do we need a whole migration for every modal state? Cardinality of these dealies has gotten to 2 - could it get to like 20? I feel like it ends up denormalizing-ish - could be just a time to say "damn, we're denormalized now, shove in a json column"

Or we could not. I think it could go either way but if there's gonna be like 6 of them that JSON column would've been the better decision. How's your crystal ball?

Reply:
The plan is to eventually make a proper system to handle this kind of thing. What we don't know at the moment if its a way to mark dismissed modals or more generalized user preferences that is much more involved. We also talked about a json column but that could end up as part of the user preferences solution.

So the idea was to just mimic the existing flag since it wouldn't add any more complexity to a migration that we will already have to do and solve it properly as its own feature rather than just grab what was needed for this fix.

You're 100% right that this isn't great but until we have a proper design this doesn't add what we don't want. It adds essentially 0 extra work without doing work we have to undo in the future.